float64 N
float64 E
float64 D
